### PR TITLE
Refactor Runs Module and Comment & Resolved state module

### PR DESCRIPTION
**Pull request by raylrui**
1. refactor runs code structure and style component
2. add comment CRUD module
image

3. add 'RESOLVED" state module
image

4. couple with https://github.com/umccr/orcabus/pull/613

### CODE REVIEW

File: src/components/common/error/DetailedErrorBoundary.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ type ErrorBoundaryProps = PropsWithChildren & { errorTitle?: string; onCloseErro
 const DetailedErrorBoundary = ({ children, errorTitle, onCloseError }: ErrorBoundaryProps) => {
   function Fallback({ error, resetErrorBoundary }: FallbackProps) {
     return (
-      <div className='my-4 relative p-4 rounded-md bg-gray-50'>
+      <div className='my-4 relative p-4 rounded-lg bg-gray-50 h-full'>
```
Comment (williamputraintan) [inline src/components/common/error/DetailedErrorBoundary.tsx:10]: Can we not have the `h-full` for now? It blocks off the entire Report Analysis table when no workflow found

e.g. http://localhost:3000/lab/library/lib.04J5M2JFE1JPYV62RYQEG99CP4/overview (in 2xl wide mode)

File: src/components/layouts/files/FilesLayout.tsx [hunk 2]
```diff
@@ -10,8 +10,8 @@ import MainArea from '../MainArea';
 const FilesLayout = ({ children }: PropsWithChildren) => {
   return (
     <MainArea>
-      <LocationBreadcrumb />
-      <Card>
+      <Card className='w-full h-full sm:px-4 lg:px-6 lg:py-4 p-2 !rounded-none'>
```
Comment (williamputraintan) [inline src/components/layouts/files/FilesLayout.tsx:13]: I guess in this case we could get rid of `<Card>` concept then, as if the card become full screen?

**Review by williamputraintan — APPROVED** [review]
A lot of changes here, and looks good to me!

File: src/components/common/error/DetailedErrorBoundary.tsx (hunk 1 above)
Comment (raylrui) [reply src/components/common/error/DetailedErrorBoundary.tsx:10]: sure, I will have a check now

File: src/utils/commonUtils.ts [hunk 3]
```diff
@@ -46,6 +57,15 @@ export function getUsername(email: string) {
     .join(' ');
 }
 
+/**
```
[the remark this answers is not in the document]
Comment (ray-lius) [reply src/utils/commonUtils.ts]: thanks, delete

File: src/components/common/error/DetailedErrorBoundary.tsx (hunk 1 above)
Comment (ray-lius) [reply src/components/common/error/DetailedErrorBoundary.tsx]: "h-full" seems redundant here, delete

File: src/components/layouts/files/FilesLayout.tsx (hunk 2 above)
Comment (ray-lius) [reply src/components/layouts/files/FilesLayout.tsx]: updated, do you want me to remove the <card> concept in labs layout as well?

Comment (williamputraintan) [reply src/components/layouts/files/FilesLayout.tsx]: That would be great! Thank you!

Comment (raylrui) [reply src/components/layouts/files/FilesLayout.tsx]: updated

**Comment by raylrui** [comment]
will merge together later  with https://github.com/umccr/orcabus/pull/613

**Pull request merged**